### DISCUSSION TITLE
Error while running SPAdes in WSL2

### DISCUSSION THREAD

**Discussion by joseailtoncruz** [ablab/spades#1159] [Q&A]
<html>
<body>
<div class="aju" style="cursor: default; align-items: center; display: flex; float: none; height: 80px; padding: 0px 16px; min-width: 40px; color: rgb(34, 34, 34); font-family: &quot;Google Sans&quot;, Roboto, RobotoDraft, Helvetica, Arial, sans-serif; font-size: medium; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="aCi" style="position: relative;"><img id=":42_1-e" name=":42" src="https://lh3.googleusercontent.com/a/AAcHTteEpJ1Hmb0BqCsbguGGG6lQEvwILb-qJ7Jp7_qYTC29cw=s40-p" class="ajn" jid="ailton.cruz@ufrpe.br" data-hovercard-id="ailton.cruz@ufrpe.br" data-name="JOSÉ SANTOS" aria-hidden="true" style="width: 40px; height: 40px; border-radius: 50%; display: block; background-color: rgb(204, 204, 204);"></div></div><div class="gs" style="margin: 0px; padding: 0px 0px 20px; width: 1037.6px; color: rgb(34, 34, 34); font-family: &quot;Google Sans&quot;, Roboto, RobotoDraft, Helvetica, Arial, sans-serif; font-size: medium; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="gE iv gt" style="font-size: 0.875rem; cursor: auto; padding: 20px 0px 0px;">

JOSÉ SANTOS <ailton.cruz@ufrpe.br> | JOSÉ SANTOS <ailton.cruz@ufrpe.br> | sex., 14 de jul., 15:05 (há 11 dias) |   |  
-- | -- | -- | -- | --
JOSÉ SANTOS <ailton.cruz@ufrpe.br>
para spades.support | para spades.support
para spades.support

</div><div id=":et"><div class="qQVYZb"></div><div class="utdU2e"></div><div class="btm"></div></div><div class=""><div class="aHl" style="margin-left: -38px;"></div><div id=":fa" tabindex="-1"></div><div id=":ev" class="ii gt" jslog="20277; u014N:xr6bB; 1:WyIjdGhyZWFkLWE6cjEzMDk4NjIxMTc3ODAzNTc2MDUiLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCxudWxsLFtdXQ..; 4:WyIjbXNnLWE6cjEzMTgxMjQ1MzgxNTg3NDU1NDEiLG51bGwsW11d" style="direction: ltr; margin: 8px 0px 0px; padding: 0px; position: relative; font-size: 0.875rem;"><div id=":eu" class="a3s aiL " style="font: small / 1.5 Arial, Helvetica, sans-serif; overflow: hidden;"><div dir="ltr"><div dir="ltr"><div dir="ltr"><div dir="ltr"><div dir="ltr"><div>I am encountering an error while running SPAdes in the Windows Subsystem for Linux 2 (WSL2) environment and would appreciate your assistance in resolving the issue.</div><div><br></div><div>Error description:</div><div>I am attempting to run SPAdes version 3.13.1 in WSL2 with the following system information:</div><div>- SPAdes version: 3.13.1</div><div>- Python version: 3.10.6</div><div>- OS: Linux-5.15.90.1-microsoft-<wbr>standard-WSL2-x86_64-with-<wbr>glibc2.35</div><div><br></div><div>I am using the following command to run SPAdes:</div><div>/usr/lib/spades/bin/spades.py -1 P27_forward_paired.fq.gz -2 P27_reverse_paired.fq.gz -o spades_assembly_all_illumina</div><div><br></div><div>However, I am receiving the following error message during the read error correction step:</div><div><div><b>0:32:20.448   240M / 3G    ERROR  K-mer Counting           (kmer_data.cpp             : 351)   The reads contain too many k-mers to fit into available memory. You need approx. 11.987GB of free RAM to assemble your dataset</b></div><div><b>== Error ==  system call for: "['/usr/lib/spades/bin/spades-<wbr>hammer', '/home/ailton_ubuntu/<wbr>bioinformatica/Trimmomatic-0.<wbr>39/spades_assembly_all_<wbr>illumina/corrected/configs/<a href="http://config.info/" target="_blank" data-saferedirecturl="https://www.google.com/url?q=http://config.info&amp;source=gmail&amp;ust=1690385072799000&amp;usg=AOvVaw0SLD9v4u5V7Vg6vq8wXDzu" style="color: rgb(17, 85, 204);">con<wbr>fig.info</a>']" finished abnormally, err code: 255</b></div><div><b>In case you have troubles running SPAdes, you can write to<span> </span><a href="mailto:spades.support@cab.spbu.ru" target="_blank" style="color: rgb(17, 85, 204);">spades.support@cab.spbu.ru</a></b><br></div><div><b>or report an issue on our GitHub repository<span> </span><a href="http://github.com/ablab/spades" target="_blank" data-saferedirecturl="https://www.google.com/url?q=http://github.com/ablab/spades&amp;source=gmail&amp;ust=1690385072799000&amp;usg=AOvVaw1fvcCmdEzojYbV8-MwXBlV" style="color: rgb(17, 85, 204);">github.com/ablab/spades</a></b></div><div><b>Please provide us with params.txt and spades.log files from the output directory.</b></div></div><div><br></div><div>I have attached the following files for your reference:</div><div>- params.txt</div><div>- spades.log</div><div><br></div><div>I would appreciate any guidance or suggestions you can provide to help resolve this issue. Please let me know if there are any additional logs or files that would be helpful for troubleshooting.</div><div><br></div><div>Thank you for your attention and support.</div><div><br></div><div>Best regards,</div><div><br></div><div><div dir="ltr" class="gmail_signature"><div dir="ltr"><div><b>Atenciosamente,</b><br></div><div dir="ltr"><b><br>José Ailton Cruz Macêdo dos Santos</b><br>Estudante de Pós-graduação em fitopatologia<br>Universidade Federal Rural de Pernambuco<br></div><div dir="ltr">Departamento de Agronomia UFRPE<br></div><div dir="ltr">Pernambuco-PE<br>Fone: (81) 986837383 OI</div><font face="arial, helvetica, sans-serif"><span style="border-collapse: collapse;">E-mail:<span> </span></span></font><a href="mailto:santos.j.a.c.m@gmail.com" target="_blank" style="color: rgb(17, 85, 204);">santos.j.a.c.m@gmail.com</a><br><span style="font-family: arial, helvetica, sans-serif; font-size: 13.3333px;">Currículo Lattes:<span> </span></span><a title="Endereço para acessar este CV:" href="https://wwws.cnpq.br/cvlattesweb/PKG_MENU.menu?f_cod=CF0D9F6E8F62F07857E12471E4AA7397#" target="_blank" data-saferedirecturl="https://www.google.com/url?q=https://wwws.cnpq.br/cvlattesweb/PKG_MENU.menu?f_cod%3DCF0D9F6E8F62F07857E12471E4AA7397%23&amp;source=gmail&amp;ust=1690385072799000&amp;usg=AOvVaw3Lldb7eJ_UnxXyLUVnsAaW" style="color: rgb(50, 110, 155); margin: 0px; padding: 0px; text-decoration: none; font-family: Tahoma, Geneva, sans-serif; font-size: 11px; font-style: normal; font-variant: normal; font-weight: normal; letter-spacing: normal; line-height: normal; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; word-spacing: 0px; background-color: rgb(225, 234, 242);"><b style="margin: 0px; padding: 0px;">http://lattes.cnpq.br/<wbr>1553284695502900</b></a></div></div></div></div></div></div></div></div></div></div></div></div>
</body>
</html>JOSÉ SANTOS <ailton.cruz@ufrpe.br>
Anexos
sex., 14 de jul., 15:05 (há 11 dias)
para spades.support

I am encountering an error while running SPAdes in the Windows Subsystem for Linux 2 (WSL2) environment and would appreciate your assistance in resolving the issue.

Error description:
I am attempting to run SPAdes version 3.13.1 in WSL2 with the following system information:
- SPAdes version: 3.13.1
- Python version: 3.10.6
- OS: Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.35

I am using the following command to run SPAdes:
/usr/lib/spades/bin/spades.py -1 P27_forward_paired.fq.gz -2 P27_reverse_paired.fq.gz -o spades_assembly_all_illumina

However, I am receiving the following error message during the read error correction step:
0:32:20.448   240M / 3G    ERROR  K-mer Counting           (kmer_data.cpp             : 351)   The reads contain too many k-mers to fit into available memory. You need approx. 11.987GB of free RAM to assemble your dataset
== Error ==  system call for: "['/usr/lib/spades/bin/spades-hammer', '/home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/spades_assembly_all_illumina/corrected/configs/[config.info](http://config.info/)']" finished abnormally, err code: 255
In case you have troubles running SPAdes, you can write to [spades.support@cab.spbu.ru](mailto:spades.support@cab.spbu.ru)
or report an issue on our GitHub repository [github.com/ablab/spades](http://github.com/ablab/spades)
Please provide us with params.txt and spades.log files from the output directory.

I have attached the following files for your reference:
- params.txt
- spades.log

I would appreciate any guidance or suggestions you can provide to help resolve this issue. Please let me know if there are any additional logs or files that would be helpful for troubleshooting.

Thank you for your attention and support.

Best regards,

Atenciosamente,

José Ailton Cruz Macêdo dos Santos
Estudante de Pós-graduação em fitopatologia
Universidade Federal Rural de Pernambuco
Departamento de Agronomia UFRPE
Pernambuco-PE
Fone: (81) 986837383 OI
E-mail: [santos.j.a.c.m@gmail.com](mailto:santos.j.a.c.m@gmail.com)
Currículo Lattes: [http://lattes.cnpq.br/1553284695502900](https://wwws.cnpq.br/cvlattesweb/PKG_MENU.menu?f_cod=CF0D9F6E8F62F07857E12471E4AA7397#)

WSL: # Settings apply across all Linux distros running on WSL 2
[wsl2]

# Limits VM memory to use no more than 4 GB, this can be set as whole numbers using GB or MB
memory=12GB 

# Sets amount of swap storage space to 8GB, default is 25% of available RAM
swap=10GB

SPADE.LOG: Command line: /usr/lib/spades/bin/spades.py	-1	/home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/P27_forward_paired.fq.gz	-2	/home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/P27_reverse_paired.fq.gz	-o	/home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/spades_assembly_all_illumina	

System information:
  SPAdes version: 3.13.1
  Python version: 3.10.6
  OS: Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.35

Output dir: /home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/spades_assembly_all_illumina
Mode: read error correction and assembling
Debug mode is turned OFF

Dataset parameters:
  Multi-cell mode (you should set '--sc' flag if input data was obtained with MDA (single-cell) technology or --meta flag if processing metagenomic dataset)
  Reads:
    Library number: 1, library type: paired-end
      orientation: fr
      left reads: ['/home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/P27_forward_paired.fq.gz']
      right reads: ['/home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/P27_reverse_paired.fq.gz']
      interlaced reads: not specified
      single reads: not specified
      merged reads: not specified
Read error correction parameters:
  Iterations: 1
  PHRED offset will be auto-detected
  Corrected reads will be compressed
Assembly parameters:
  k: automatic selection based on read length
  Repeat resolution is enabled
  Mismatch careful mode is turned OFF
  MismatchCorrector will be SKIPPED
  Coverage cutoff is turned OFF
Other parameters:
  Dir for temp files: /home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/spades_assembly_all_illumina/tmp
  Threads: 16
  Memory limit (in Gb): 9

======= SPAdes pipeline started. Log can be found here: /home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/spades_assembly_all_illumina/spades.log

===== Read error correction started. 

== Running read error correction tool: /usr/lib/spades/bin/spades-hammer /home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/spades_assembly_all_illumina/corrected/configs/config.info

  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from N/A, git revision N/A
  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/spades_assembly_all_illumina/corrected/configs/config.info
  0:00:00.014     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 12
  0:00:00.015     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 9 Gb
  0:00:00.015     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset
  0:00:00.015     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33
  0:00:00.015     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]
  0:00:00.015     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes
     === ITERATION 0 begins ===
  0:00:00.016     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index
  0:00:00.016     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 192 files using 12 threads. This might take a while.
  0:00:00.016     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024
  0:00:00.016     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.249891 Gb
  0:00:00.016     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 174686
  0:00:00.363     3G / 3G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/P27_forward_paired.fq.gz
  0:00:16.308     3G / 3G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1277388 reads
  0:00:27.416     3G / 3G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2574638 reads
  0:00:45.084     3G / 3G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 3894827 reads
  0:00:55.600     3G / 3G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 5194453 reads
  0:01:06.433     3G / 3G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 6504286 reads
  0:01:17.793     3G / 3G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 7834471 reads
  0:01:34.915     3G / 3G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 9136749 reads
  0:01:45.419     3G / 3G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 10431629 reads
  0:01:56.158     3G / 3G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 11733306 reads
  0:02:56.505     3G / 3G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 16954423 reads
  0:05:36.467     3G / 3G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/P27_reverse_paired.fq.gz
  0:06:40.406     3G / 3G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 34323060 reads
  0:11:42.636     3G / 3G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 58197810 reads processed
  0:11:42.791    48M / 3G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.
  0:30:20.880    48M / 3G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 321772420 kmers in total.
  0:30:20.880    48M / 3G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.
  0:30:43.278    48M / 3G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices
  0:31:09.590   240M / 3G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.
  0:31:52.099   240M / 3G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 149214408 bytes occupied (3.70981 bits per kmer).
  0:32:20.448   240M / 3G    ERROR  K-mer Counting           (kmer_data.cpp             : 351)   The reads contain too many k-mers to fit into available memory. You need approx. 11.987GB of free RAM to assemble your dataset

== Error ==  system call for: "['/usr/lib/spades/bin/spades-hammer', '/home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/spades_assembly_all_illumina/corrected/configs/config.info']" finished abnormally, err code: 255

In case you have troubles running SPAdes, you can write to spades.support@cab.spbu.ru
or report an issue on our GitHub repository github.com/ablab/spades
Please provide us with params.txt and spades.log files from the output directory.

PARAMS: Command line: /usr/lib/spades/bin/spades.py	-1	/home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/P27_forward_paired.fq.gz	-2	/home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/P27_reverse_paired.fq.gz	-o	/home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/spades_assembly_all_illumina	

System information:
  SPAdes version: 3.13.1
  Python version: 3.10.6
  OS: Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.35

Output dir: /home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/spades_assembly_all_illumina
Mode: read error correction and assembling
Debug mode is turned OFF

Dataset parameters:
  Multi-cell mode (you should set '--sc' flag if input data was obtained with MDA (single-cell) technology or --meta flag if processing metagenomic dataset)
  Reads:
    Library number: 1, library type: paired-end
      orientation: fr
      left reads: ['/home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/P27_forward_paired.fq.gz']
      right reads: ['/home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/P27_reverse_paired.fq.gz']
      interlaced reads: not specified
      single reads: not specified
      merged reads: not specified
Read error correction parameters:
  Iterations: 1
  PHRED offset will be auto-detected
  Corrected reads will be compressed
Assembly parameters:
  k: automatic selection based on read length
  Repeat resolution is enabled
  Mismatch careful mode is turned OFF
  MismatchCorrector will be SKIPPED
  Coverage cutoff is turned OFF
Other parameters:
  Dir for temp files: /home/ailton_ubuntu/bioinformatica/Trimmomatic-0.39/spades_assembly_all_illumina/tmp
  Threads: 16
  Memory limit (in Gb): 9

**Comment by asl** [comment]
The log reads:
```
0:32:20.448   240M / 3G    ERROR  K-mer Counting           (kmer_data.cpp             : 351)   The reads contain too many k-mers to fit into available memory. You need approx. 11.987GB of free RAM to assemble your dataset
```

So you need more RAM to assemble your dataset. You machine has only 9 that is not enough.
